indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -204,15 +204,17 @@ public TaskStatus run(TaskToolbox toolbox) throws Exception
     }
 
     final String specVersion = indexerSchema.getTuningConfig().getVersion();
-    if (indexerSchema.getTuningConfig().isUseExplicitVersion() && version.compareTo(specVersion) > 0) {
-      version = specVersion;
-    } else {
-      log.error(
-          "Spec version can not be greater than lock version, Spec version: [%s] Lock version: [%s].",
-          specVersion,
-          version
-      );
-      return TaskStatus.failure(getId());
+    if (indexerSchema.getTuningConfig().isUseExplicitVersion()) {
+      if (specVersion.compareTo(version) < 0) {
```

Comment:
I think version cannot be null, see this - https://github.com/druid-io/druid/blob/master/indexing-hadoop/src/main/java/io/druid/indexer/HadoopTuningConfig.java#L115